src/instrumentation.rs

```diff
@@ -0,0 +1,342 @@
+use async_trait::async_trait;
+use dashmap::DashMap;
+use dipstick::{Input, InputScope, Prefixed, Statsd, StatsdScope};
+use log::{debug, error, info, warn};
+use rdkafka::message::{BorrowedMessage, Message};
+use std::convert::TryInto;
+use std::sync::Arc;
+use std::time::Instant;
+use tokio::{
+    sync::mpsc::{channel, Receiver, Sender},
+    task,
+};
+
+pub type Statistic = (StatTypes, i64);
+
+pub fn init_stats(endpoint: &str, app_id: &str) -> Result<Arc<Sender<Statistic>>, std::io::Error> {
+    let scope = Statsd::send_to(endpoint)?.named(app_id).metrics();
+    let scope_ref = Arc::new(scope);
+
+    let mut handler = StatsHandler::new(scope_ref.clone());
+
+    let channel = Arc::new(handler.tx.clone());
```

Comment:
Yeah, since the spawn moves the stats handler, I need to get a clone of the sender before that. Naming of this var was also incorrect. Changed to:

```
let sender = handler.tx.clone();
```